chain/evm/compass.go

```diff
@@ -841,6 +847,91 @@ func (t *compass) GetSendToPalomaEvents(ctx context.Context, orchestrator string
 	return events, err
 }
 
+func (t *compass) GetLightNodeSaleEvents(ctx context.Context, orchestrator string) ([]chain.LightNodeSaleEvent, error) {
+	filter, err := ethfilter.Factory().
+		WithFromBlockNumberProvider(t.evm.FindCurrentBlockNumber).
+		WithFromBlockNumberSafetyMargin(1).
+		WithTopics([]common.Hash{crypto.Keccak256Hash([]byte(nodeSaleEvent))}).
+		WithAddresses(t.smartContractAddr).
+		Filter(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	currentBlockNumber := filter.FromBlock.Int64()
+	if t.lastObservedBlockHeights.lightNodeSaleEvent == 0 {
+		t.lastObservedBlockHeights.lightNodeSaleEvent = currentBlockNumber - 10_000
+	}
+
+	filter.FromBlock = big.NewInt(t.lastObservedBlockHeights.lightNodeSaleEvent)
+	filter.ToBlock = big.NewInt(min(t.lastObservedBlockHeights.lightNodeSaleEvent+10_000, currentBlockNumber))
```

Comment:
We no longer have monotonic behavior now that the nonce is shared across 2 events. .

Since we're relying on the skyway nonce here, is it maybe better to combine this with the gravity events scanner? If we have both event types emitted within a few blocks of each other, and either of these scans find the event with the higher nonce first, the second one will be disregarded.

Reply:
That makes sense, I'll check it.

But, just to confirm, don't we have that problem already with `BatchSendEvent` and `SendToPalomaEvent`?

Reply:
Yes, yes we do...

Reply:
Maybe it's best so slightly rewrite this whole event observation into only one handler, which applies multiple filters for the same query window.

What do you think?

Reply:
Yes, I agree that's better. I'll do that.